### PR TITLE
Add optional resourcePrefix param for Chart

### LINKED ISSUE #520: Duplicate resource URN with helm when using same release name on multiple clusters
I create several kubernetes clusters in the same stack which all have some core helm charts installed (traefik, external-dns...). I want the release name to be the same on all clusters but the helm chart resource only uses the release name as the name of the resource.
With any other resource, I can provide the resource name so I could just use  `` `${provider}/${region}/${cluster}/${releasename}` ``  to solve this. It's not possible right now since I can only provide the release name.

### PR DESCRIPTION
**Pull request by lblackstone**
Allow users to disambiguate resources generated from a helm
Chart using the optional `resourcePrefix` parameter. If set, the
generated resources will be prefixed with the provided string.

Fixes #520

### CODE REVIEW

File: pkg/gen/python-templates/helm/v2/helm.py [hunk 1]
```diff
@@ -273,11 +290,14 @@ def __init__(self,
         :param Optional[List[Callable]] transformations: Optional list of transformations to apply to
                resources that will be created by this chart prior to creation. Allows customization of the
                chart behaviour without directly modifying the chart itself.
+        :param Optional[str] resource_prefix: An optional prefix for the auto-generated resource names.
+               Example: A resource created with resource_prefix="foo" would produce a resource named "foo-resourceName".
```
Comment (metral) [inline pkg/gen/python-templates/helm/v2/helm.py]: Is there any way we can minimize the repetition here? We describe the `resource_prefix` param in this `helm.py` 4 times.

File: tests/examples/examples_test.go [hunk 2]
```diff
@@ -157,7 +169,7 @@ func TestExamples(t *testing.T) {
 				t *testing.T, stackInfo integration.RuntimeValidationStackInfo,
 			) {
 				assert.NotNil(t, stackInfo.Deployment)
-				assert.Equal(t, 9, len(stackInfo.Deployment.Resources))
+				assert.Equal(t, 15, len(stackInfo.Deployment.Resources))
```
Comment (metral) [inline tests/examples/examples_test.go]: We can punt this off to another PR, but i've noticed this hard-coding of numbers to expect in assertions before and IIRC we've talked about it, but I was wondering if we can move away to a computed value instead.

File: tests/examples/helm-local/index.ts [hunk 3]
```diff
@@ -54,3 +54,39 @@ const nginx = new k8s.helm.v2.Chart("simple-nginx-local", {
...
+        // Put every resource in the created namespace.
+        (obj: any) => {
+            if (obj.metadata !== undefined) {
+                obj.metadata.namespace = namespaceName
+            } else {
+                obj.metadata = {namespace: namespaceName}
+            }
+        }
+    ],
+    resourcePrefix: "dup"
```
Comment (metral) [inline tests/examples/helm-local/index.ts]: nit: can we move this prop to the top of the def? i missed it at first as it seemed buried under the `transformations`

Comment (metral) [reply tests/examples/helm-local/index.ts]: also, given that this dupe chart is the same as the orig, can we wrap this in a func that takes a `resourcePrefix` to minimize the LoC?

File: tests/examples/python/python_test.go [hunk 4]
```diff
@@ -139,11 +139,11 @@ func TestYaml(t *testing.T) {
 			// first ConfigFile.
 
 			// Verify the provider resource.
-			provRes := stackInfo.Deployment.Resources[22]
+			provRes := stackInfo.Deployment.Resources[12]
 			assert.True(t, providers.IsProviderType(provRes.URN.Type()))
 
 			// Verify root resource.
-			stackRes := stackInfo.Deployment.Resources[23]
+			stackRes := stackInfo.Deployment.Resources[13]
```
Comment (metral) [inline tests/examples/python/python_test.go]: Same number vs computed value question, as mentioned above.

File: tests/examples/python/python_test.go [hunk 5]
```diff
@@ -288,7 +288,8 @@ func TestHelm(t *testing.T) {
 		t.FailNow()
 	}
 	options := baseOptions.With(integration.ProgramTestOptions{
-		Dir: filepath.Join(cwd, "helm"),
+		Dir:                  filepath.Join(cwd, "helm"),
+		ExpectRefreshChanges: true,
```
Comment (metral) [inline tests/examples/python/python_test.go]: What triggered the expectancy of `refreshChanges` here?

File: tests/integration/yaml-url/step1/index.ts [hunk 6]
```diff
@@ -39,15 +39,15 @@ new k8s.yaml.ConfigFile("guestbook", {
...
+          obj.metadata.namespace = namespace2.metadata.name;
+        } else {
+          obj.metadata = {namespace: namespace2.metadata.name}
+        }
       }
     }
-  }
-],
+  ],
   resourcePrefix: "dup"
```
Comment (metral) [inline tests/integration/yaml-url/step1/index.ts]: Same question as above about moving this prop up to the top of the def

File: pkg/gen/python-templates/helm/v2/helm.py (hunk 1 above)
Comment (lblackstone) [reply pkg/gen/python-templates/helm/v2/helm.py]: I created a base class to cut down on the repetition slightly.

File: tests/examples/examples_test.go (hunk 2 above)
Comment (lblackstone) [reply tests/examples/examples_test.go]: Yeah, I'll punt for now. It's something I have in mind for test cleanup.

File: tests/examples/python/python_test.go (hunk 5 above)
Comment (lblackstone) [reply tests/examples/python/python_test.go]: This chart includes a `PodDisruptionBudget` that has a status update once the Deployment rolls out. This causes changes on refresh. It _would_ be nice to have an option to ignore only status changes, but there's currently no way to do that with the test runner.

File: pkg/gen/python-templates/helm/v2/helm.py [hunk 7]
```diff
@@ -191,6 +181,37 @@ class ChartOpts:
...
+               chart behaviour without directly modifying the chart itself.
+        :param Optional[str] resource_prefix: An optional prefix for the auto-generated resource names.
+               Example: A resource created with resource_prefix="foo" would produce a resource named "foo-resourceName".
+        """
+        self.namespace = namespace
+        self.values = values
+        self.transformations = transformations
+        self.resource_prefix = resource_prefix
+
+
```
Comment (metral) [inline pkg/gen/python-templates/helm/v2/helm.py]: nit: rm extra line

**Pull request merged**